Tests/ImperialTests/ImperialTests.swift

```diff
@@ -1,10 +1,218 @@
+    }
+
+    @Test("GitHub Route")
+    func githubRoute() async throws {
+        try await withApp { app in
+            try await app.test(
+                .GET, "/github",
+                afterResponse: { res async throws in
+                    #expect(res.status == .notFound)
+                })
+
+            try app.oAuth(from: GitHub.self, authenticate: "github", callback: "gh-auth-complete", redirect: "/")
+
+            try await app.test(
+                .GET, "/github",
+                afterResponse: { res async throws in
+                    #expect(res.status != .notFound)
+                })
+        }
+    }
+
+    @Test("Gitlab Route")
+    func gitlabRoute() async throws {
+        try await withApp { app in
+            try await app.test(
+                .GET, "/gitlab",
+                afterResponse: { res async throws in
+                    #expect(res.status == .notFound)
+                })
+
+            try app.oAuth(from: Gitlab.self, authenticate: "gitlab", callback: "gitlab-auth-complete", redirect: "/")
+
+            try await app.test(
+                .GET, "/gitlab",
+                afterResponse: { res async throws in
+                    #expect(res.status != .notFound)
+                })
+        }
+    }
+
+    @Test("Google Route")
+    func googleRoute() async throws {
+        try await withApp { app in
+            try await app.test(
+                .GET, "/google",
+                afterResponse: { res async throws in
+                    #expect(res.status == .notFound)
+                })
+
+            try app.oAuth(from: Google.self, authenticate: "google", callback: "google-auth-complete", redirect: "/")
+
+            try await app.test(
+                .GET, "/google",
+                afterResponse: { res async throws in
+                    #expect(res.status != .notFound)
+                })
+        }
+    }
+
+    @Test("Google JWT Route")
+    func googleJWTRoute() async throws {
+        try await withApp { app in
+            try await app.test(
+                .GET, "/googleJWT",
+                afterResponse: { res async throws in
+                    #expect(res.status == .notFound)
+                })
+
+            try app.oAuth(from: GoogleJWT.self, authenticate: "googleJWT", callback: "googleJWT-auth-complete", redirect: "/")
+
+            try await app.test(
+                .GET, "/googleJWT",
+                afterResponse: { res async throws in
+                    #expect(res.status != .notFound)
+                })
+        }
+    }
+
+    @Test("Keycloak Route")
+    func keycloakRoute() async throws {
+        try await withApp { app in
+            try await app.test(
+                .GET, "/keycloak",
+                afterResponse: { res async throws in
+                    #expect(res.status == .notFound)
+                })
+
+            try app.oAuth(from: Keycloak.self, authenticate: "keycloak", callback: "keycloak-auth-complete", redirect: "/")
```

Comment:
Editing routes after the app is initialised is not ideal and something we shouldn't be showcasing